app/tests/integration/nimbus/test_e2e_create_experiment.py

```diff
@@ -126,3 +129,100 @@ def test_create_new_experiment_remote_settings(selenium, base_url):
+    overview.select_risk_revenue_false()
+    overview.select_risk_partner_false()
+
+    # Fill Branches page
+    branches = overview.save_and_continue()
+    branches.remove_branch()
+    branches.reference_branch_name = "name 1"
+    branches.reference_branch_description = "a nice experiment"
+    branches.feature_config = "No Feature Firefox Desktop"
+
+    # Fill Metrics page
+    metrics = branches.save_and_continue()
+
+    # Fill Audience page
+    audience = metrics.save_and_continue()
+    audience.channel = "Nightly"
+    audience.min_version = 80
+    audience.targeting = "US_ONLY"
+    audience.percentage = 50.0
+    audience.expected_clients = 50
+    audience.save_btn()
+    review = audience.save_and_continue()
+
+    # Review and approve
+    selenium.find_element_by_css_selector("#PageRequestReview")
+    review.launch_without_preview.click()
+    review.request_review.click_launch_checkboxes()
+    review.request_review.request_launch_button.click()
+    review.approve()
+    selenium.get("http://kinto:8888/v1/admin")
+    kinto_login = Login(selenium, base_url).wait_for_page_to_load()
+    kinto_login.kinto_auth.click()
+    kinto_dashbard = kinto_login.login()
+    bucket = kinto_dashbard.buckets[-1]
+    for item in bucket.bucket_category:
+        if "nimbus-desktop-experiments" in item.text:
+            item.click()
+            break
+    record = kinto_dashbard.record
+    record.action("reject")
+    kinto_dashbard = Dashboard(selenium, base_url)
+    modal = kinto_dashbard.reject_modal
+    modal.decline_changes()
+
+    # Load home page and wait for experiment to show in the Drafts tab
+    selenium.get(base_url)
+    for attempt in range(45):
+        try:
+            home = HomePage(selenium, base_url).wait_for_page_to_load()
+            home.tabs[-1].click()
+            new_experiments = len(home.tables[0].experiments)
+            assert new_experiments != current_experiments
+        except AssertionError:
+            time.sleep(2)
+            selenium.refresh()
+        else:
+            home.tabs[-1].click()
+            draft_experiments = home.tables[0]
+            experiment_found = False
+            for item in draft_experiments.experiments:
+                if experiment_name in item.text:
+                    item.click()
+                    experiment_found = True
+                    break
+            if experiment_found:
+                break
+        if attempt == 45:
+            raise AssertionError("Experiment was not found")
+    experiment_url = experiment_name.replace(" ", "-")
+    selenium.get(f"{base_url}/{experiment_url}/request-review")
+    for attempt in range(30):
+        try:
+            summary_page = ReviewPage(selenium, base_url).wait_for_page_to_load()
+            assert summary_page.rejected_text, "Rejected text box did not load"
+        except NoSuchElementException:
+            time.sleep(2)
+            selenium.refresh()
+        if attempt == 30:
```

Comment:
OK, some lessons learned:

```python
attempts = range(4)
for attempt in attempts:
  if attempt == attempts.start:
    print("First iteration, who dis?")
  print(attempt)
  if attempt == attempts.stop - 1:
    print("END")
```

### OUTPUT

```
First iteration, who dis?
0
1
2
3
END
```

So `range(4)` loops from 0..3, which makes sense; but it confused me for a bit since the maximum `attempt` value (3) never matched `attempts.stop` value (4), so I had to use `-1` to detect the last iteration.
Also, it looks like `range(4)` returns a `range` instance, where as my `attempt` value is an integer, so I can't use something like `attempt.first` since that's ... not how any of this works. So I declared the `range(4)` outside the `for..in` loop because I couldn't think of another way.

Although, further Binging shows I can do a `for..in` with an `else` that might work (although arguably not entirely intuitive, but I'm also an idiot):

```python
for attempt in range(4):
  print(attempt)
else:
  print("END")
```

### OUTPUT

```
0
1
2
3
END
```

So, in summary, not sure if `attempt` will ever equal the specified target value for `range()`, or if we need to subtract 1. Also, my use cases were super simple, so might not account for nested `try..except` and other real world logic.

Thanks for coming to my TED talk.

Reply:
I had it at range -1 but changed it. This is how it should be.

Reply:
I tried a smaller test case which is closer to the current implementation and I don't think my `if attempt == 45:` is working.

```python
attempts = range(5)
for attempt in attempts:
    try:
        assert 1 != 1
    except AssertionError:
        print(f"AssertionError; retrying {attempt} of {attempts.stop}")
    else:
        print("success!")
        break # exit the `for..in` loop.

    if attempt == 5:
        raise AssertionError("Experiment was not found")
else:
    print("for..else")
```

### OUTPUT

```
AssertionError; retrying 0 of 5
AssertionError; retrying 1 of 5
AssertionError; retrying 2 of 5
AssertionError; retrying 3 of 5
AssertionError; retrying 4 of 5
for..else
```

Note that it prints the debug output from the `except AssertionError:` block, but never seems to hit the `raise AssertionError("Experiment was not found")` clause.

Raising the assertion in the `else` block worked, or using `attempts.stop - 1`

```python
    if attempt == attempts.stop - 1:
        raise AssertionError("Experiment was not found")
```